core/src/main/java/greencity/controller/EcoNewsCommentController.java

```diff
@@ -0,0 +1,224 @@
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import springfox.documentation.annotations.ApiIgnore;
+
+@Validated
+@AllArgsConstructor
+@RestController
+@RequestMapping("/econews/comments")
+public class EcoNewsCommentController {
+    private final EcoNewsCommentService ecoNewsCommentService;
+    private final UserService userService;
+    private final EcoNewsService ecoNewsService;
+
+    /**
+     * Method for creating {@link greencity.entity.EcoNewsComment}.
+     *
+     * @param econewsId id of {@link greencity.entity.EcoNews} to add comment to.
+     * @param addEcoNewsCommentDtoRequest - dto for {@link greencity.entity.EcoNewsComment} entity.
+     * @return dto {@link greencity.dto.econewscomment.AddEcoNewsCommentDtoResponse}
+     */
+    @ApiOperation(value = "Add comment.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = HttpStatuses.CREATED, response = CommentReturnDto.class),
+        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @PostMapping("{econewsId}")
+    public ResponseEntity<Object> save(@PathVariable Long econewsId,
+                                       @Valid @RequestBody AddEcoNewsCommentDtoRequest addEcoNewsCommentDtoRequest,
+                                       @ApiIgnore @AuthenticationPrincipal Principal principal) {
+        User user = userService.findByEmail(principal.getName());
+
+        return ResponseEntity
+            .status(HttpStatus.CREATED)
+            .body(ecoNewsCommentService.save(econewsId, addEcoNewsCommentDtoRequest, user));
+    }
+
+    /**
+     * Method to get all comments to {@link greencity.entity.EcoNews} specified by ecoNewsId.
+     *
+     * @param ecoNewsId id of {@link greencity.entity.EcoNews}
+     * @return Pageable of {@link EcoNewsCommentDto}
+     */
+    @ApiOperation(value = "Get all comments.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("")
+    @ApiPageable
+    public ResponseEntity<PageableDto<EcoNewsCommentDto>> findAll(@ApiIgnore Pageable pageable,
+                                                                  Long ecoNewsId,
+                                                                  @ApiIgnore @AuthenticationPrincipal
+                                                                      Principal principal) {
+        User user = null;
+        if (principal != null) {
```

Comment:
Is is possible that principal is null? (I have heard this question one of Academy demos)

Reply:
> Is is possible that principal is null? (I have heard this question one of Academy demos)
 Yes. If user is not authenticated (didn't sign in), principal is null